Index: boinc-5.10.8/client/Makefile.am
===================================================================
--- boinc-5.10.8.orig/client/Makefile.am	2007-06-26 08:55:11.000000000 +0200
+++ boinc-5.10.8/client/Makefile.am	2007-06-26 08:57:25.000000000 +0200
@@ -66,7 +66,7 @@
 boinc_client_LDADD = -L../lib -lboinc $(PTHREAD_LIBS)
 
 boinc_clientdir = $(bindir)
-boinc_client_DATA = ../curl/ca-bundle.crt
+#boinc_client_DATA = ../curl/ca-bundle.crt
 
 switcher_SOURCES = switcher.C
 
