Index: boinc-5.10.25/client/Makefile.am
===================================================================
--- boinc-5.10.25.orig/client/Makefile.am
+++ boinc-5.10.25/client/Makefile.am
@@ -66,7 +66,7 @@
 boinc_client_LDADD = -L../lib -lboinc $(PTHREAD_LIBS)
 
 boinc_clientdir = $(bindir)
-boinc_client_DATA = ../curl/ca-bundle.crt
+#boinc_client_DATA = ../curl/ca-bundle.crt
 
 switcher_SOURCES = switcher.C
 switcher_CPPFLAGS = -O3 -fomit-frame-pointer -fforce-addr -ffast-math $(AM_CPPFLAGS)
