Index: boinc-5.10.8/client/Makefile.am
===================================================================
--- boinc-5.10.8.orig/client/Makefile.am	2007-06-26 08:57:25.000000000 +0200
+++ boinc-5.10.8/client/Makefile.am	2007-06-26 08:57:32.000000000 +0200
@@ -61,7 +61,7 @@
     work_fetch.C
 
 boinc_client_DEPENDENCIES = $(LIBRSA)
-boinc_client_CPPFLAGS = -O3 -fomit-frame-pointer -fforce-addr -ffast-math $(AM_CPPFLAGS)
+boinc_client_CPPFLAGS = $(AM_CPPFLAGS)
 boinc_client_LDFLAGS = -static-libgcc
 boinc_client_LDADD = -L../lib -lboinc $(PTHREAD_LIBS)
 
