Index: boinc-5.10.0/Makefile.am
===================================================================
--- boinc-5.10.0.orig/Makefile.am	2006-08-31 23:16:36.000000000 +0200
+++ boinc-5.10.0/Makefile.am	2007-05-30 08:59:43.000000000 +0200
@@ -17,7 +17,7 @@
 endif
 
 if BUILD_CLIENTGUI
-   CLIENTGUI_SUBDIRS = clientgui sea
+   CLIENTGUI_SUBDIRS = clientgui
 endif
 
 # ORDER MATTERS below.  One must build dependencies FIRST, then things
