--- boinc-4.72.orig/configure.ac	2005-07-23 02:52:41.000000000 +0200
+++ boinc-4.72/configure.ac	2005-08-08 18:30:36.000000000 +0200
@@ -396,21 +396,12 @@
 
 dnl check for wxWidgets
 
-if test "${enable_client_release}" = yes ; then
-    if test "${enable_debug}" = yes ; then
-        AM_PATH_WXCONFIG([2.6.0], [wxWin=1], [wxWin=0], --static=yes --unicode=no --debug=yes)
-    else
-        AM_PATH_WXCONFIG([2.6.0], [wxWin=1], [wxWin=0], --static=yes --unicode=no --debug=no)
-    fi
+if test "${enable_debug}" = yes ; then
+    AM_PATH_WXCONFIG([2.6.0], [wxWin=1], [wxWin=0], --static=no --unicode=yes --debug=yes)
 else
-    if test "${enable_debug}" = yes ; then
-        AM_PATH_WXCONFIG([2.6.0], [wxWin=1], [wxWin=0], --unicode=no --debug=yes)
-    else
-        AM_PATH_WXCONFIG([2.6.0], [wxWin=1], [wxWin=0], --unicode=no --debug=no)
-    fi
+    AM_PATH_WXCONFIG([2.6.0], [wxWin=1], [wxWin=0], --static=no --unicode=yes --debug=no)
 fi
 
-
 if ( test "${enable_client}" = yes ) && ( test "$wxWin" != 1 ); then
    AC_MSG_WARN([
 ================================================================================
