Index: boinc-5.10.25/client/cs_platforms.C
===================================================================
--- boinc-5.10.25.orig/client/cs_platforms.C
+++ boinc-5.10.25/client/cs_platforms.C
@@ -93,6 +93,8 @@
 
 #endif
 
+    if (strlen(config.alt_platform)>0) add_platform(config.alt_platform);
+
     if (config.no_alt_platform) {
         PLATFORM p = platforms[0];
         platforms.clear();
Index: boinc-5.10.25/client/log_flags.C
===================================================================
--- boinc-5.10.25.orig/client/log_flags.C
+++ boinc-5.10.25/client/log_flags.C
@@ -219,6 +219,7 @@
         if (xp.parse_bool(tag, "disallow_attach", disallow_attach)) continue;
         if (xp.parse_bool(tag, "os_random_only", os_random_only)) continue;
         if (xp.parse_bool(tag, "no_alt_platform", no_alt_platform)) continue;
+        if (xp.parse_str(tag, "alt_platform", alt_platform, sizeof(alt_platform))) continue;
         if (xp.parse_bool(tag, "simple_gui_only", simple_gui_only)) continue;
         if (xp.parse_bool(tag, "dont_contact_ref_site", dont_contact_ref_site)) continue;
         if (xp.parse_str(tag, "data_dir", path, sizeof(path))) {
Index: boinc-5.10.25/client/log_flags.h
===================================================================
--- boinc-5.10.25.orig/client/log_flags.h
+++ boinc-5.10.25/client/log_flags.h
@@ -92,6 +92,7 @@
     bool disallow_attach;
     bool os_random_only;
     bool no_alt_platform;
+    char alt_platform[256];
     bool simple_gui_only;
     bool dont_contact_ref_site;
 
