Index: boinc-5.10.30/db/db_base.h
===================================================================
--- boinc-5.10.30.orig/db/db_base.h	2007-02-13 22:41:07.000000000 +0100
+++ boinc-5.10.30/db/db_base.h	2007-12-20 05:53:11.000000000 +0100
@@ -20,6 +20,9 @@
 #ifndef _DB_BASE_
 #define _DB_BASE_
 
+#include <cstdlib>
+#include <string>
+
 #include <mysql.h>
 
 // if SQL columns are not 'not null', you must use these safe_atoi, safe_atof
Index: boinc-5.10.30/sched/edf_sim.C
===================================================================
--- boinc-5.10.30.orig/sched/edf_sim.C	2007-05-10 23:50:52.000000000 +0200
+++ boinc-5.10.30/sched/edf_sim.C	2007-12-20 05:53:11.000000000 +0100
@@ -19,6 +19,7 @@
 
 #include <vector>
 #include <algorithm>
+#include <cstdio>
 
 #include "edf_sim.h"
 
Index: boinc-5.10.30/sched/edf_sim.h
===================================================================
--- boinc-5.10.30.orig/sched/edf_sim.h	2007-05-10 18:23:10.000000000 +0200
+++ boinc-5.10.30/sched/edf_sim.h	2007-12-20 05:53:11.000000000 +0100
@@ -20,6 +20,7 @@
 #ifndef _EDF_SIM_H
 #define _EDF_SIM_H
 
+#include <cstring>
 #include <vector>
 
 struct IP_RESULT {
Index: boinc-5.10.30/sched/validator.C
===================================================================
--- boinc-5.10.30.orig/sched/validator.C	2007-05-11 18:30:13.000000000 +0200
+++ boinc-5.10.30/sched/validator.C	2007-12-20 05:53:11.000000000 +0100
@@ -37,6 +37,7 @@
 
 #include "config.h"
 #include <unistd.h>
+#include <climits>
 #include <cmath>
 #include <vector>
 
Index: boinc-5.10.30/sched/sched_locality.C
===================================================================
--- boinc-5.10.30.orig/sched/sched_locality.C	2007-05-31 20:14:45.000000000 +0200
+++ boinc-5.10.30/sched/sched_locality.C	2007-12-20 05:53:11.000000000 +0100
@@ -39,6 +39,8 @@
 #include "sched_locality.h"
 #include "sched_util.h"
 
+#include <algorithm>
+#include <climits>
 #include <vector>
 #include <string>
 #include <cstring>
Index: boinc-5.10.30/lib/crypt.C
===================================================================
--- boinc-5.10.30.orig/lib/crypt.C	2006-12-19 21:49:12.000000000 +0100
+++ boinc-5.10.30/lib/crypt.C	2007-12-20 05:53:11.000000000 +0100
@@ -26,6 +26,7 @@
 #include <cctype>
 #include <cstdio>
 #include <cstdlib>
+#include <cstring>
 #endif
 
 #include "md5_file.h"
Index: boinc-5.10.30/lib/parse.h
===================================================================
--- boinc-5.10.30.orig/lib/parse.h	2007-08-18 02:45:03.000000000 +0200
+++ boinc-5.10.30/lib/parse.h	2007-12-20 05:53:11.000000000 +0100
@@ -25,6 +25,7 @@
 #else
 #include <cstdio>
 #include <cstdlib>
+#include <cstring>
 #include <string>
 #include <math.h>
 #ifdef solaris
Index: boinc-5.10.30/lib/procinfo_unix.C
===================================================================
--- boinc-5.10.30.orig/lib/procinfo_unix.C	2007-03-21 00:12:22.000000000 +0100
+++ boinc-5.10.30/lib/procinfo_unix.C	2007-12-20 05:53:11.000000000 +0100
@@ -29,6 +29,7 @@
 
 #include <stdio.h>
 
+#include <cstring>
 #include <ctype.h>
 #include <sys/types.h>
 #include <dirent.h>
Index: boinc-5.10.30/client/cs_account.C
===================================================================
--- boinc-5.10.30.orig/client/cs_account.C	2007-05-14 21:18:45.000000000 +0200
+++ boinc-5.10.30/client/cs_account.C	2007-12-20 05:53:11.000000000 +0100
@@ -23,6 +23,7 @@
 
 #ifndef _WIN32
 #include "config.h"
+#include <algorithm>
 #include <cstdio>
 #include <cassert>
 #ifdef HAVE_SYS_STAT_H
Index: boinc-5.10.30/client/client_msgs.C
===================================================================
--- boinc-5.10.30.orig/client/client_msgs.C	2006-06-22 21:40:30.000000000 +0200
+++ boinc-5.10.30/client/client_msgs.C	2007-12-20 05:53:11.000000000 +0100
@@ -22,6 +22,7 @@
 #else
 #include "config.h"
 #include <cstdarg>
+#include <cstring>
 #include <deque>
 #endif
 
