Index: boinc/clientgui/AdvancedFrame.cpp
===================================================================
--- boinc.orig/clientgui/AdvancedFrame.cpp	2011-04-25 18:41:15.000000000 +0200
+++ boinc/clientgui/AdvancedFrame.cpp	2011-04-25 18:43:41.000000000 +0200
@@ -1492,14 +1492,14 @@
     int prog;
 #endif
     int argc = 3;
-    char* argv[3];
+    char* const argv[3] = {
+	 const_cast<char *>("boincmgr"),
+	 const_cast<char *>("--multiple"),
+	 const_cast<char *>("")
+    };
 
     wxString strExecutable = wxGetApp().GetRootDirectory() + wxGetApp().GetExecutableName();
 
-    argv[0] = "boincmgr";
-    argv[1] = "--multiple";
-    argv[2] = "";
-
     run_program(
         wxGetApp().GetRootDirectory().mb_str(),
         strExecutable.mb_str(),
@@ -1508,6 +1508,7 @@
         2.0,
         prog
     );
+
 #else
     char s[512];
     unsigned char procName[256];
