Index: boinc/clientgui/res/skins/default/graphic/advanced_link_image.xpm
===================================================================
--- boinc.orig/clientgui/res/skins/default/graphic/advanced_link_image.xpm	2011-04-09 16:01:00.000000000 +0200
+++ boinc/clientgui/res/skins/default/graphic/advanced_link_image.xpm	2011-04-09 16:01:15.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char *advanced_link_image_xpm[] = {
+static const char *advanced_link_image_xpm[] = {
 /* columns rows colors chars-per-pixel */
 "101 20 67 1",
 "o c Black",
