Author: Rom Walton <romw@romwnet.org>
Bug-Debian: http://bugs.debian.org/641593

Index: boinc/clientgui/ProjectInfoPage.cpp
===================================================================
--- boinc.orig/clientgui/ProjectInfoPage.cpp	2012-05-30 23:05:52.000000000 +0800
+++ boinc/clientgui/ProjectInfoPage.cpp	2012-05-30 23:05:55.000000000 +0800
@@ -488,6 +488,7 @@
 void CProjectInfoPage::OnProjectSelected( wxCommandEvent& WXUNUSED(event) ) {
     wxLogTrace(wxT("Function Start/End"), wxT("CProjectInfoPage::OnProjectSelected - Function Begin"));
 
+    if (m_pProjectsCtrl->GetSelection() != wxNOT_FOUND) {
     CProjectInfo* pProjectInfo = (CProjectInfo*)m_pProjectsCtrl->GetClientData(m_pProjectsCtrl->GetSelection());
 
     wxString strURL = pProjectInfo->m_strURL;
@@ -539,6 +540,7 @@
     // Set tooltip to full text in case ellipsed
     m_pProjectDetailsOrganizationCtrl->SetToolTip(pProjectInfo->m_strOrganization);
 
+    }
     wxLogTrace(wxT("Function Start/End"), wxT("CProjectInfoPage::OnProjectSelected - Function End"));
 }
 
