Author: Gianfranco Costamagna costamagnagianfranco@yahoo.it

Origin: upstream, 9506b117 http://boinc.berkeley.edu/trac/changeset/9506b117a1df4d5754d891ec12cb8aa76d4dd192/boinc
Forwarded: not-needed
Reviewed-By: costamagnagianfranco@yahoo.it
Last-Update: <2012-11-01>

--- boinc-7.0.36+dfsg.orig/lib/filesys.h
+++ boinc-7.0.36+dfsg/lib/filesys.h
@@ -35,7 +35,7 @@
 #endif
 
 #ifndef MAXPATHLEN
-#define MAXPATHLEN 1024
+#define MAXPATHLEN 4096
 #endif
 
 #endif /* !WIN32 */
--- boinc-7.0.36+dfsg.orig/lib/gui_rpc_client.h
+++ boinc-7.0.36+dfsg/lib/gui_rpc_client.h
@@ -33,13 +33,14 @@
 #include <locale.h>
 #endif
 
-#include "miofile.h"
-#include "prefs.h"
-#include "hostinfo.h"
+#include "cc_config.h"
 #include "common_defs.h"
-#include "notice.h"
+#include "filesys.h"
+#include "hostinfo.h"
+#include "miofile.h"
 #include "network.h"
-#include "cc_config.h"
+#include "notice.h"
+#include "prefs.h"
 
 struct GUI_URL {
     std::string name;
--- boinc-7.0.36+dfsg.orig/lib/boinc_win.h
+++ boinc-7.0.36+dfsg/lib/boinc_win.h
@@ -22,10 +22,6 @@
 #ifndef _BOINC_WIN_
 #define _BOINC_WIN_
 
-#ifndef MAXPATHLEN
-#define MAXPATHLEN 4096
-#endif
-
 #ifndef __CYGWIN32__
 
 // Windows C Runtime Library
--- boinc-7.0.36+dfsg.orig/lib/common_defs.h
+++ boinc-7.0.36+dfsg/lib/common_defs.h
@@ -191,5 +191,4 @@ struct VERSION_INFO {
 #define DEFAULT_SS_EXECUTABLE       "boincscr"
 #endif
 
-
 #endif
