Author: Steffen Moeller <moeller@debian.org>
Description: Increasing buffers for technical descriptions

Index: boinc/client/client_types.h
===================================================================
--- boinc.orig/client/client_types.h	2012-03-31 13:47:58.636220955 +0200
+++ boinc/client/client_types.h	2012-03-31 13:51:15.653530626 +0200
@@ -814,12 +814,12 @@
 
     int coproc_indices[MAX_COPROCS_PER_JOB];
         // keep track of coprocessor reservations
-    char resources[1024];
+    char resources[256];
         // textual description of resources used
     double schedule_backoff;
         // don't try to schedule until this time
         // (wait for free GPU RAM)
-    char schedule_backoff_reason[1024];
+    char schedule_backoff_reason[256];
 };
 
 // represents an always/auto/never value, possibly temporarily overridden
Index: boinc/client/client_types.cpp
===================================================================
--- boinc.orig/client/client_types.cpp	2012-03-31 13:47:01.692998711 +0200
+++ boinc/client/client_types.cpp	2012-03-31 13:51:15.661530518 +0200
@@ -182,7 +182,7 @@
 // parse project fields from client_state.xml
 //
 int PROJECT::parse_state(XML_PARSER& xp) {
-    char buf[256];
+    char buf[1024];
     std::string sched_url, stemp;
     string str1, str2;
     int retval, rt;
