Index: boinc/api/gutil.cpp
===================================================================
--- boinc.orig/api/gutil.cpp	2012-08-18 14:30:37.274778529 +0200
+++ boinc/api/gutil.cpp	2012-08-18 14:42:31.653112311 +0200
@@ -24,7 +24,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <setjmp.h>
 
 #ifdef _WIN32
Index: boinc/api/gutil_text.cpp
===================================================================
--- boinc.orig/api/gutil_text.cpp	2012-08-17 16:08:20.402579016 +0200
+++ boinc/api/gutil_text.cpp	2012-08-18 14:42:31.657112257 +0200
@@ -26,7 +26,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <setjmp.h>
 
 #ifdef _WIN32
Index: boinc/client/client_state.cpp
===================================================================
--- boinc.orig/client/client_state.cpp	2012-08-18 14:42:31.265117561 +0200
+++ boinc/client/client_state.cpp	2012-08-18 14:42:31.661112203 +0200
@@ -26,7 +26,7 @@
 #include <ctime>
 #include <cstdarg>
 #include <cstring>
-#include <math.h>
+#include <cmath>
 #if HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
Index: boinc/client/sim_util.cpp
===================================================================
--- boinc.orig/client/sim_util.cpp	2012-08-17 16:08:20.402579016 +0200
+++ boinc/client/sim_util.cpp	2012-08-18 14:42:31.665112149 +0200
@@ -35,7 +35,7 @@
 #define finite _finite
 #endif
 
-#include <math.h>
+#include <cmath>
 
 #include "error_numbers.h"
 #include "str_util.h"
Index: boinc/client/whetstone.cpp
===================================================================
--- boinc.orig/client/whetstone.cpp	2012-08-17 16:08:20.402579016 +0200
+++ boinc/client/whetstone.cpp	2012-08-18 14:42:31.669112094 +0200
@@ -29,7 +29,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <time.h>
 #endif
 
Index: boinc/clientgui/sqlite3.c
===================================================================
--- boinc.orig/clientgui/sqlite3.c	2012-08-17 16:08:20.406578962 +0200
+++ boinc/clientgui/sqlite3.c	2012-08-18 14:42:31.737111174 +0200
@@ -20222,7 +20222,7 @@
 **
 */
 #ifdef SQLITE_HAVE_ISNAN
-# include <math.h>
+# include <cmath>
 #endif
 
 /*
Index: boinc/clientscr/ss_app.cpp
===================================================================
--- boinc.orig/clientscr/ss_app.cpp	2012-08-18 14:30:37.430776418 +0200
+++ boinc/clientscr/ss_app.cpp	2012-08-18 14:42:31.741111121 +0200
@@ -22,7 +22,7 @@
 #ifdef _WIN32
 #include "boinc_win.h"
 #else
-#include <math.h>
+#include <cmath>
 #endif
 #include <string>
 #include <vector>
Index: boinc/db/boinc_db.cpp
===================================================================
--- boinc.orig/db/boinc_db.cpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/db/boinc_db.cpp	2012-08-18 14:42:31.761110850 +0200
@@ -22,7 +22,7 @@
 #include <ctime>
 #include <unistd.h>
 #include <cmath>
-#include <math.h>
+#include <cmath>
 
 // For machines with finite() defined in ieeefp.h
 #if HAVE_IEEEFP_H
Index: boinc/lib/boinc_win.h
===================================================================
--- boinc.orig/lib/boinc_win.h	2012-08-17 16:08:20.406578962 +0200
+++ boinc/lib/boinc_win.h	2012-08-18 14:42:31.809110201 +0200
@@ -197,7 +197,7 @@
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
-#include <math.h>
+#include <cmath>
 #include <setjmp.h>
 #include <signal.h>
 #include <stdarg.h>
Index: boinc/lib/md5.c
===================================================================
--- boinc.orig/lib/md5.c	2012-08-17 16:08:20.406578962 +0200
+++ boinc/lib/md5.c	2012-08-18 14:42:31.841109767 +0200
@@ -88,7 +88,7 @@
  * For reference, here is the program that computed the T values.
  */
 #if 0
-#include <math.h>
+#include <cmath>
 main()
 {
     int i;
Index: boinc/lib/parse.cpp
===================================================================
--- boinc.orig/lib/parse.cpp	2012-08-18 14:42:31.041120592 +0200
+++ boinc/lib/parse.cpp	2012-08-18 14:42:31.877109279 +0200
@@ -31,7 +31,7 @@
 #include <cstring>
 #include <cstdlib>
 #include <string>
-#include <math.h>
+#include <cmath>
 #include <ctype.h>
 #include <errno.h>
 #if HAVE_IEEEFP_H
Index: boinc/lib/str_util.cpp
===================================================================
--- boinc.orig/lib/str_util.cpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/lib/str_util.cpp	2012-08-18 14:42:31.913108793 +0200
@@ -27,7 +27,7 @@
 #ifndef _WIN32
 #include "config.h"
 #include <string>
-#include <math.h>
+#include <cmath>
 #include <string.h>
 #include <stdlib.h>
 #include <ctype.h>
Index: boinc/lib/util.cpp
===================================================================
--- boinc.orig/lib/util.cpp	2012-08-18 14:42:31.313116911 +0200
+++ boinc/lib/util.cpp	2012-08-18 14:42:31.945108359 +0200
@@ -44,7 +44,7 @@
 #include <errno.h>
 #include <string>
 #include <cstring>
-#include <math.h>
+#include <cmath>
 #if HAVE_IEEEFP_H
 #include <ieeefp.h>
 extern "C" {
Index: boinc/samples/atiopencl/atiopencl.hpp
===================================================================
--- boinc.orig/samples/atiopencl/atiopencl.hpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/samples/atiopencl/atiopencl.hpp	2012-08-18 14:42:31.965108089 +0200
@@ -28,7 +28,7 @@
 #endif
 
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <time.h>
 #include <cstdlib>
 #include <iostream>
Index: boinc/samples/example_app/uc2_graphics.cpp
===================================================================
--- boinc.orig/samples/example_app/uc2_graphics.cpp	2012-08-18 14:30:37.462775986 +0200
+++ boinc/samples/example_app/uc2_graphics.cpp	2012-08-18 14:42:31.973107981 +0200
@@ -33,7 +33,7 @@
 #ifdef _WIN32
 #include "boinc_win.h"
 #else
-#include <math.h>
+#include <cmath>
 #endif
 
 #include "parse.h"
Index: boinc/samples/nvcuda/cuda_kernel.cu
===================================================================
--- boinc.orig/samples/nvcuda/cuda_kernel.cu	2012-08-17 16:08:20.406578962 +0200
+++ boinc/samples/nvcuda/cuda_kernel.cu	2012-08-18 14:42:31.977107927 +0200
@@ -25,7 +25,7 @@
 // identity matrix and the product A * incerse(A). For debugging...
 //#define VERIFY 1
 #include <stdio.h>
-#include <math.h>
+#include <cmath>
 #include <time.h>
 #include "cuda_config.h"
 
Index: boinc/samples/nvopencl/nvopencl.hpp
===================================================================
--- boinc.orig/samples/nvopencl/nvopencl.hpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/samples/nvopencl/nvopencl.hpp	2012-08-18 14:42:31.981107873 +0200
@@ -24,7 +24,7 @@
 #include <CL/cl.h>
 #include <oclUtils.h>
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <time.h>
 #include <cstdlib>
 #include <iostream>
Index: boinc/samples/vboxwrapper/vboxwrapper.cpp
===================================================================
--- boinc.orig/samples/vboxwrapper/vboxwrapper.cpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/samples/vboxwrapper/vboxwrapper.cpp	2012-08-18 14:42:31.985107818 +0200
@@ -56,7 +56,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <stdio.h>
-#include <math.h>
+#include <cmath>
 #include <string>
 #include <unistd.h>
 #endif
Index: boinc/samples/wrappture/fermi.cpp
===================================================================
--- boinc.orig/samples/wrappture/fermi.cpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/samples/wrappture/fermi.cpp	2012-08-18 14:42:32.021107331 +0200
@@ -2,7 +2,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
-#include <math.h>
+#include <cmath>
 #include <unistd.h>
 
 int main(int argc, char * argv[]) {
Index: boinc/samples/wrappture/wrappture_example.cpp
===================================================================
--- boinc.orig/samples/wrappture/wrappture_example.cpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/samples/wrappture/wrappture_example.cpp	2012-08-18 14:42:32.029107223 +0200
@@ -15,7 +15,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
-#include <math.h>
+#include <cmath>
 #ifndef _WIN32
 #include <unistd.h>
 #endif
Index: boinc/sched/credit.cpp
===================================================================
--- boinc.orig/sched/credit.cpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/sched/credit.cpp	2012-08-18 14:42:32.033107169 +0200
@@ -22,7 +22,7 @@
 // Note: this is credit.cpp rather than sched_credit.cpp
 // because you might grant credit e.g. from a trickle handler
 
-#include <math.h>
+#include <cmath>
 
 #include "boinc_db.h"
 #include "error_numbers.h"
Index: boinc/sched/feeder.cpp
===================================================================
--- boinc.orig/sched/feeder.cpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/sched/feeder.cpp	2012-08-18 14:42:32.037107115 +0200
@@ -98,7 +98,7 @@
 #include <ctime>
 #include <csignal>
 #include <unistd.h>
-#include <math.h>
+#include <cmath>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/param.h>
Index: boinc/vda/vda_lib.cpp
===================================================================
--- boinc.orig/vda/vda_lib.cpp	2012-08-17 16:08:20.406578962 +0200
+++ boinc/vda/vda_lib.cpp	2012-08-18 14:42:32.041107061 +0200
@@ -23,7 +23,7 @@
 #include <vector>
 #include <set>
 #include <algorithm>
-#include <math.h>
+#include <cmath>
 #include <limits.h>
 
 using std::vector;
