Author: Steffen Moeller <moeller@debian.org>
Description: Initialisation.

Index: boinc/lib/diagnostics_win.cpp
===================================================================
--- boinc.orig/lib/diagnostics_win.cpp	2012-02-16 01:37:40.000000000 +0100
+++ boinc/lib/diagnostics_win.cpp	2012-02-16 10:59:49.000000000 +0100
@@ -329,8 +329,9 @@
 
     do {
         *ppBuffer = HeapAlloc(hHeap, HEAP_ZERO_MEMORY, *pcbBuffer);
-        if (ppBuffer == NULL) {
+        if (*ppBuffer == NULL) {
             retval = ERROR_NOT_ENOUGH_MEMORY;
+            return retval;
         }
 
         Status = pNTQSI(
@@ -1630,6 +1631,7 @@
             break;
         case EXCEPTION_ACCESS_VIOLATION:
             strcpy(status, "Access Violation");
+            substatus[0]=0;
             if (pExPtrs->ExceptionRecord->NumberParameters == 2) {
                 switch(pExPtrs->ExceptionRecord->ExceptionInformation[0]) {
                 case 0: // read attempt
Index: boinc/lib/crypt.cpp
===================================================================
--- boinc.orig/lib/crypt.cpp	2012-02-15 22:04:05.000000000 +0100
+++ boinc/lib/crypt.cpp	2012-02-16 10:54:04.000000000 +0100
@@ -339,6 +339,7 @@
     char clear_buf[MD5_LEN];
     int n, retval;
     DATA_BLOCK clear_signature;
+    clear_buf[0]=0;
 
     n = (int)strlen(md5_buf);
     clear_signature.data = (unsigned char*)clear_buf;
