Author: Steffen Moeller <moeller@debian.org>
Description: Path length beyond 256.

Index: boinc/client/app_start.cpp
===================================================================
--- boinc.orig/client/app_start.cpp	2012-03-24 16:04:03.000000000 +0100
+++ boinc/client/app_start.cpp	2012-03-24 16:12:00.000000000 +0100
@@ -122,7 +122,7 @@
 static void coproc_cmdline(
     int rsc_type, RESULT* rp, double ninstances, char* cmdline
 ) {
-    char buf[256];
+    char buf[1024];
     COPROC* coproc = &coprocs.coprocs[rsc_type];
     for (int j=0; j<ninstances; j++) {
         int k = rp->coproc_indices[j];
@@ -159,7 +159,7 @@
     if (!shm_handle) return ERR_SHMGET;
     sprintf(shmem_seg_name, "boinc_%d", i);
 #else
-    char init_data_path[256];
+    char init_data_path[1024];
 #ifndef __EMX__
     // shmem_seg_name is not used with mmap() shared memory
     if (app_version->api_major_version() >= 6) {
@@ -194,7 +194,7 @@
 }
 
 void ACTIVE_TASK::init_app_init_data(APP_INIT_DATA& aid) {
-    char project_dir[256], project_path[256];
+    char project_dir[1024], project_path[1024];
 
     aid.major_version = BOINC_MAJOR_VERSION;
     aid.minor_version = BOINC_MINOR_VERSION;
@@ -278,7 +278,7 @@
 //
 int ACTIVE_TASK::write_app_init_file(APP_INIT_DATA& aid) {
     FILE *f;
-    char init_data_path[256];
+    char init_data_path[1024];
 
     sprintf(init_data_path, "%s/%s", slot_dir, INIT_DATA_FILE);
 
@@ -352,7 +352,7 @@
 int ACTIVE_TASK::setup_file(
     FILE_INFO* fip, FILE_REF& fref, char* file_path, bool input, bool is_io_file
 ) {
-    char link_path[256], rel_file_path[256], open_name[256];
+    char link_path[1024], rel_file_path[1024], open_name[256];
     int retval;
     PROJECT* project = result->project;
 
@@ -438,7 +438,7 @@
 }
 
 int ACTIVE_TASK::copy_output_files() {
-    char slotfile[256], projfile[256], open_name[256];
+    char slotfile[1024], projfile[1024], open_name[1024];
     unsigned int i;
     for (i=0; i<result->output_files.size(); i++) {
         FILE_REF& fref = result->output_files[i];
@@ -480,7 +480,7 @@
 //   ACTIVE_TASK::task_state is PROCESS_EXECUTING
 //
 int ACTIVE_TASK::start() {
-    char exec_name[256], file_path[256], buf[256], exec_path[256];
+    char exec_name[1024], file_path[1024], buf[1024], exec_path[1024];
     char cmdline[80000];    // 64KB plus some extra
     unsigned int i;
     FILE_REF fref;
@@ -627,7 +627,7 @@
     PROCESS_INFORMATION process_info;
     STARTUPINFO startup_info;
     LPVOID environment_block = NULL;
-    char slotdirpath[256];
+    char slotdirpath[1024];
     char error_msg[1024];
     char error_msg2[1024];
 
Index: boinc/client/auto_update.cpp
===================================================================
--- boinc.orig/client/auto_update.cpp	2012-03-18 14:31:22.000000000 +0100
+++ boinc/client/auto_update.cpp	2012-03-24 16:09:25.000000000 +0100
@@ -179,7 +179,7 @@
     unsigned int i;
     FILE_INFO* fip=0;
     char version_dir[1024];
-    char cwd[256];
+    char cwd[1024];
     char *argv[10];
     int retval, argc;
 #ifdef _WIN32
