--- a/sched/sched_send.cpp
+++ b/sched/sched_send.cpp
@@ -1432,7 +1432,7 @@
 
     if (g_request->coprocs.nvidia.count && ssp->have_cuda_apps) {
         send_gpu_property_messages(cuda_requirements,
-            g_request->coprocs.nvidia.prop.dtotalGlobalMem,
+            g_request->coprocs.nvidia.prop.totalGlobalMem,
             g_request->coprocs.nvidia.display_driver_version,
             "NVIDIA GPU"
         );
