--- a/clientgui/DlgAbout.cpp
+++ b/clientgui/DlgAbout.cpp
@@ -78,12 +78,14 @@
     m_AboutBOINCLogoCtrl = NULL;
     m_AboutBOINCSloganCtrl = NULL;
     m_AboutBOINCURLCtrl = NULL;
-#if defined(_WIN64) && defined(_M_X64)
+#if defined(__x86_64__) || defined(__x86_64) || defined(_M_AMD64)
     m_strVersion.Printf(wxT("%s (x64)"), wxT(BOINC_VERSION_STRING));
+#elif defined(__i386__) || defined(__i386) || defined(_M_IX86)
+    m_strVersion.Printf(wxT("%s (x86)"), wxT(BOINC_VERSION_STRING));
 #elif defined(__ppc__)
     m_strVersion.Printf(wxT("%s (PowerPC)"), wxT(BOINC_VERSION_STRING));
 #else
-    m_strVersion.Printf(wxT("%s (x86)"), wxT(BOINC_VERSION_STRING));
+    m_strVersion.Printf(wxT("%s (unknown)"), wxT(BOINC_VERSION_STRING));
 #endif
     m_strWidgetsVersion.Printf(wxT("%d.%d.%d"), wxMAJOR_VERSION, wxMINOR_VERSION, wxRELEASE_NUMBER);
 ////@end CDlgAbout member initialisation
