Index: boinc/lib/parse.cpp
===================================================================
--- boinc.orig/lib/parse.cpp	2012-03-18 13:57:51.000000000 +0100
+++ boinc/lib/parse.cpp	2012-03-18 14:18:55.000000000 +0100
@@ -179,9 +179,10 @@
         int n = (int)strlen(line);
         if (nused + n >= bufsize) {
             bufsize *= 2;
-            char *b = (char*)realloc(buf, bufsize);
-            if (!b) {
-                free(buf);
+            char *b = buf;
+            buf = (char*)realloc(b, bufsize);
+            if (!buf) {
+                free(b); // b was buf and was allocated in this fun
                 return ERR_XML_PARSE;
             }
         }
