Index: boinc/clientgui/Makefile.am
===================================================================
--- boinc.orig/clientgui/Makefile.am	2011-10-31 21:51:49.000000000 +0100
+++ boinc/clientgui/Makefile.am	2011-10-31 21:52:36.000000000 +0100
@@ -47,6 +47,7 @@
     BOINCBaseFrame.cpp \
     BOINCBaseView.cpp \
     BOINCBaseWizard.cpp \
+    BOINCBitmapComboBox.cpp \
     BOINCClientManager.cpp \
     BOINCDialupManager.cpp \
     BOINCGUIApp.cpp \
