Index: boinc/sched/sched_customize.cpp
===================================================================
--- boinc.orig/sched/sched_customize.cpp	2010-12-05 16:39:02.000000000 +0100
+++ boinc/sched/sched_customize.cpp	2010-12-05 16:42:57.000000000 +0100
@@ -108,7 +108,7 @@
             return PLAN_REJECT_ATI_NO_DEVICE;
         }
 
-        hu.flops = cp->flops();
+        hu.flops = cp->peak_flops();
 
         // assume we'll need 0.5% as many CPU FLOPS as GPU FLOPS
         // to keep the GPU fed.
@@ -187,7 +187,7 @@
             }
             return PLAN_REJECT_CUDA_MEM;
         }
-        hu.flops = cp->flops_estimate();
+        hu.flops = cp->peak_flops();
         if (!strcmp(plan_class, "cuda23")) {
             hu.flops *= 1.01;
         }
