Index: boinc/generate_svn_version.sh
===================================================================
--- boinc.orig/generate_svn_version.sh	2010-12-11 22:41:48.000000000 +0100
+++ boinc/generate_svn_version.sh	2010-12-11 22:44:17.000000000 +0100
@@ -7,7 +7,9 @@
 echo "#define SVN_VERSION_H" >> $TMPFILE
 echo "" >> $TMPFILE
 
-if [ -d .git ]; then
+if [ -d debian ]; then
+    CMD=""
+elif [ -d .git ]; then
     CMD="git svn info"
 elif [ -d .svn ]; then
     CMD="svn info"
