Author: Steffen Moeller <moeller@debian.org>
Description: Helping verbosity of error messages.

Index: boinc/lib/filesys.cpp
===================================================================
--- boinc.orig/lib/filesys.cpp	2012-03-31 13:47:20.020748355 +0200
+++ boinc/lib/filesys.cpp	2012-03-31 13:47:27.320648639 +0200
@@ -136,7 +136,10 @@
     dirp->handle = INVALID_HANDLE_VALUE;
 #else
     dirp = opendir(p);
-    if (!dirp) return NULL;
+    if (!dirp) {
+        fprintf(stderr,"dir_open: Could not open directory '%s'.\n",p);
+        return NULL;
+    }
 #endif
     return dirp;
 }
