Index: boinc/lib/Makefile.am
===================================================================
--- boinc.orig/lib/Makefile.am	2010-12-05 16:21:17.000000000 +0100
+++ boinc/lib/Makefile.am	2010-12-05 16:21:26.000000000 +0100
@@ -208,5 +208,5 @@
 msg_test_LDADD = $(LIBBOINC)
 crypt_prog_SOURCES = crypt_prog.cpp 
 crypt_prog_CXXFLAGS = $(PTHREAD_CFLAGS)
-crypt_prog_LDADD = $(LIBBOINC) $(LIBBOINC_CRYPT_STATIC) $(SSL_LIBS) 
+crypt_prog_LDADD = $(LIBBOINC_CRYPT_STATIC) $(LIBBOINC) $(SSL_LIBS) 
 
