Index: boinc/lib/mac_address.cpp
===================================================================
--- boinc.orig/lib/mac_address.cpp	2012-03-03 01:12:22.000000000 +0100
+++ boinc/lib/mac_address.cpp	2012-03-03 01:14:00.000000000 +0100
@@ -144,6 +144,10 @@
 
 bool
 get_mac_addresses(char* addresses) {
+    if (!addresses) {
+        perror("get_mac_addresses: 'addresses' parameter was NULL");
+        return false;
+    }
 #if defined(_WIN32)
     IP_ADAPTER_INFO AdapterInfo[16]; // Allocate information for up to 16 NICs
 	DWORD dwBufLen = sizeof(AdapterInfo); // Save memory size of buffer
@@ -264,6 +268,10 @@
         }
         hw_addr=(struct ether_addr *)&(item->lifr_lifru.lifru_enaddr);  
 #endif
+        if (!hw_addr) {
+            perror("get_mac_addresses: Could not determine hw_addr");
+            return false;
+        }
         strcat(addresses, delimiter);
         delimiter[0] = ':';
         delimiter[1] = '\0';
