Author: Steffen Moeller <moeller@debian.org>
Description: Adding attribute 'const' to attributes of MainDocument
Index: boinc/clientgui/MainDocument.h
===================================================================
--- boinc.orig/clientgui/MainDocument.h	2012-03-30 23:24:22.722662278 +0200
+++ boinc/clientgui/MainDocument.h	2012-03-30 23:25:30.993729219 +0200
@@ -247,13 +247,13 @@
     double                      m_fResultsRPCExecutionTime;
     wxDateTime                  m_dtKillInactiveGfxTimestamp;
     std::vector<RUNNING_GFX_APP> m_running_gfx_apps;
-    RUNNING_GFX_APP*            GetRunningGraphicsApp(RESULT* result, int slot);
+    RUNNING_GFX_APP*            GetRunningGraphicsApp(RESULT* result, const int slot);
     void                        KillAllRunningGraphicsApps();
     void                        KillInactiveGraphicsApps();
 #ifdef _WIN32
     void                        KillGraphicsApp(HANDLE pid);
 #else
-    void                        KillGraphicsApp(int tpid);
+    void                        KillGraphicsApp(const int tpid);
 #endif
 
 public:
Index: boinc/clientgui/MainDocument.cpp
===================================================================
--- boinc.orig/clientgui/MainDocument.cpp	2012-03-30 23:25:30.837731351 +0200
+++ boinc/clientgui/MainDocument.cpp	2012-03-30 23:25:30.997729165 +0200
@@ -1630,7 +1630,7 @@
     kill_program(pid);
 }
 #else
-void CMainDocument::KillGraphicsApp(int pid) {
+void CMainDocument::KillGraphicsApp(const int pid) {
     char currentDir[1024];
     char thePIDbuf[20];
     int id, iRetVal;
--- boinc.orig/lib/util.cpp	2012-03-30 23:42:06.944079056 +0200
+++ boinc/lib/util.cpp	2012-03-30 23:42:09.248047487 +0200
@@ -376,7 +376,9 @@
 
 #ifdef _WIN32
 int run_program(
-    const char* dir, const char* file, int argc, char *const argv[], double nsecs, HANDLE& id
+    const char* const dir, const char* const file, const int argc,
+    /* not const */ char *const argv[],
+    const double nsecs, HANDLE& id
 ) {
     int retval;
     PROCESS_INFORMATION process_info;
@@ -428,7 +430,9 @@
 }
 #else
 int run_program(
-    const char* dir, const char* file, int , char *const argv[], double nsecs, int& id
+    const char* const dir, const char* const file, const int,
+    /* not const */ char *const argv[],
+    const double nsecs, int& id
 ) {
     int retval;
     int pid = fork();
@@ -458,7 +462,7 @@
     TerminateProcess(pid, 0);
 }
 #else
-void kill_program(int pid) {
+void kill_program(const int pid) {
     kill(pid, SIGKILL);
 }
 #endif
@@ -486,12 +490,12 @@
 }
 
 #else
-int get_exit_status(int pid) {
+int get_exit_status(const int pid) {
     int status;
     waitpid(pid, &status, 0);
     return status;
 }
-bool process_exists(int pid) {
+bool process_exists(const int pid) {
     int p = waitpid(pid, 0, WNOHANG);
     if (p == pid) return false;     // process has exited
     if (p == -1) return false;      // PID doesn't exist
@@ -530,7 +534,7 @@
     return 0;
 }
 
-int wait_client_mutex(const char* dir, double timeout) {
+int wait_client_mutex(const char* dir, const double timeout) {
     double start = dtime();
     int retval = 0;
     while (1) {
@@ -542,7 +546,7 @@
     return retval;
 }
 
-bool boinc_is_finite(double x) {
+bool boinc_is_finite(const double x) {
 #if defined (HPUX_SOURCE)
     return _Isfinite(x);
     return false;
Index: boinc/lib/util.h
===================================================================
--- boinc.orig/lib/util.h	2012-03-30 23:42:06.944079056 +0200
+++ boinc/lib/util.h	2012-03-30 23:42:09.252047432 +0200
@@ -83,7 +83,9 @@
 #ifdef _WIN32
 
 extern int run_program(
-    const char* dir, const char* file, int argc, char *const argv[], double, HANDLE&
+    const char* const dir, const char* const file, const int argc,
+    /* not const */ char *const argv[],
+    const double, HANDLE&
 );
 
 extern void kill_program(HANDLE);
@@ -92,14 +94,16 @@
 
 #else
 extern int run_program(
-    const char* dir, const char* file, int argc, char *const argv[], double, int&
+    const char* const dir, const char* const file, const int argc,
+    /* not const */ char *const argv[],
+    const double, int&
 );
-extern void kill_program(int);
-extern int get_exit_status(int);
-extern bool process_exists(int);
+extern void kill_program(const int);
+extern int get_exit_status(const int);
+extern bool process_exists(const int);
 #endif
 
-extern int wait_client_mutex(const char* dir, double timeout);
+extern int wait_client_mutex(const char* dir, const double timeout);
 
 #ifdef GCL_SIMULATOR
 extern double simtime;
