Index: boinc/clientgui/MainDocument.h
===================================================================
--- boinc.orig/clientgui/MainDocument.h	2012-03-30 23:24:22.722662278 +0200
+++ boinc/clientgui/MainDocument.h	2012-03-30 23:25:30.993729219 +0200
@@ -247,13 +247,13 @@
     double                      m_fResultsRPCExecutionTime;
     wxDateTime                  m_dtKillInactiveGfxTimestamp;
     std::vector<RUNNING_GFX_APP> m_running_gfx_apps;
-    RUNNING_GFX_APP*            GetRunningGraphicsApp(RESULT* result, int slot);
+    RUNNING_GFX_APP*            GetRunningGraphicsApp(RESULT* result, const int slot);
     void                        KillAllRunningGraphicsApps();
     void                        KillInactiveGraphicsApps();
 #ifdef _WIN32
     void                        KillGraphicsApp(HANDLE pid);
 #else
-    void                        KillGraphicsApp(int tpid);
+    void                        KillGraphicsApp(const int tpid);
 #endif
 
 public:
Index: boinc/clientgui/MainDocument.cpp
===================================================================
--- boinc.orig/clientgui/MainDocument.cpp	2012-03-30 23:25:30.837731351 +0200
+++ boinc/clientgui/MainDocument.cpp	2012-03-30 23:25:30.997729165 +0200
@@ -1630,7 +1630,7 @@
     kill_program(pid);
 }
 #else
-void CMainDocument::KillGraphicsApp(int pid) {
+void CMainDocument::KillGraphicsApp(const int pid) {
     char currentDir[1024];
     char thePIDbuf[20];
     int id, iRetVal;
