Index: boinc/Makefile.am
===================================================================
--- boinc.orig/Makefile.am	2011-04-27 22:39:13.000000000 +0200
+++ boinc/Makefile.am	2011-04-27 22:45:42.000000000 +0200
@@ -21,10 +21,10 @@
 endif
 
 if ENABLE_MANAGER
-	CLIENTGUI_SUBDIRS = clientgui clientgui/res locale
-	if BUILD_X11_SCREENSAVER
-		CLIENTGUI_SUBDIRS += clientscr
-	endif 
+CLIENTGUI_SUBDIRS = clientgui clientgui/res locale
+##if BUILD_X11_SCREENSAVER
+##	CLIENTGUI_SUBDIRS += clientscr
+##endif 
 endif
 
 # ORDER MATTERS below.  One must build dependencies FIRST, then things
