Index: boinc/Makefile.am
===================================================================
--- boinc.orig/Makefile.am	2010-12-12 01:18:18.000000000 +0100
+++ boinc/Makefile.am	2010-12-12 01:20:09.000000000 +0100
@@ -21,10 +21,10 @@
 endif
 
 if ENABLE_MANAGER
-	CLIENTGUI_SUBDIRS = clientgui clientgui/res locale
-	if BUILD_X11_SCREENSAVER
-		CLIENTGUI_SUBDIRS += clientscr
-	endif 
+CLIENTGUI_SUBDIRS = clientgui clientgui/res locale
+##if BUILD_X11_SCREENSAVER
+##	CLIENTGUI_SUBDIRS += clientscr
+##endif 
 endif
 
 # ORDER MATTERS below.  One must build dependencies FIRST, then things
