Index: boinc/tools/Makefile.am
===================================================================
--- boinc.orig/tools/Makefile.am	2012-07-08 20:36:42.483643292 +0200
+++ boinc/tools/Makefile.am	2012-07-08 20:36:51.027526800 +0200
@@ -24,7 +24,7 @@
 		run_in_ops
 
 AM_CXXFLAGS += $(MYSQL_CFLAGS)
-AM_LDFLAGS += -static
+AM_LDFLAGS += -static -L../sched -L../lib/.libs
 
 cancel_jobs_SOURCES = cancel_jobs.cpp
 cancel_jobs_LDADD = $(SERVERLIBS)
Index: boinc/vda/Makefile.am
===================================================================
--- boinc.orig/vda/Makefile.am	2012-07-08 20:36:42.483643292 +0200
+++ boinc/vda/Makefile.am	2012-07-08 20:36:51.035526690 +0200
@@ -3,7 +3,7 @@
 bin_PROGRAMS = vda vdad ssim
 
 AM_CXXFLAGS += $(MYSQL_CFLAGS)
-AM_LDFLAGS += -static
+AM_LDFLAGS += -static -L../lib/.libs -L../sched
 
 vda_SOURCES = vda.cpp vda_policy.cpp
 vda_LDADD = $(SERVERLIBS)
Index: boinc/sched/Makefile.am
===================================================================
--- boinc.orig/sched/Makefile.am	2012-07-08 20:36:42.483643292 +0200
+++ boinc/sched/Makefile.am	2012-07-08 20:36:51.039526636 +0200
@@ -4,7 +4,7 @@
 include $(top_srcdir)/Makefile.incl
 
 AM_CPPFLAGS += $(MYSQL_CFLAGS) $(PTHREAD_CFLAGS)
-AM_LDFLAGS += -static
+AM_LDFLAGS += -static -L../lib/.libs/
 
 if ENABLE_LIBRARIES
 
Index: boinc/apps/Makefile.am
===================================================================
--- boinc.orig/apps/Makefile.am	2012-07-08 20:36:42.483643292 +0200
+++ boinc/apps/Makefile.am	2012-07-08 20:36:51.043526582 +0200
@@ -8,15 +8,15 @@
 
 upper_case_SOURCES = upper_case.cpp
 upper_case_CXXFLAGS = $(PTHREAD_CFLAGS)
-upper_case_LDFLAGS = -static-libtool-libs $(PTHREAD_CFLAGS)
+upper_case_LDFLAGS = -static-libtool-libs $(PTHREAD_CFLAGS) -L../api/.libs -L../lib/.libs
 upper_case_LDADD = $(APPLIBS)
 
 concat_SOURCES = concat.cpp
 concat_CXXFLAGS = $(PTHREAD_CFLAGS)
-concat_LDFLAGS = -static-libtool-libs $(PTHREAD_CFLAGS)
+concat_LDFLAGS = -static-libtool-libs $(PTHREAD_CFLAGS) -L../api/.libs -L../lib/.libs
 concat_LDADD = $(APPLIBS)
 
 1sec_SOURCES = 1sec.cpp
 1sec_CXXFLAGS = $(PTHREAD_CFLAGS)
-1sec_LDFLAGS = -static-libtool-libs $(PTHREAD_CFLAGS)
+1sec_LDFLAGS = -static-libtool-libs $(PTHREAD_CFLAGS) -L../api/.libs -L../lib/.libs
 1sec_LDADD = $(APPLIBS)
Index: boinc/clientgui/Makefile.am
===================================================================
--- boinc.orig/clientgui/Makefile.am	2012-07-08 20:10:13.000000000 +0200
+++ boinc/clientgui/Makefile.am	2012-07-08 20:38:08.338472678 +0200
@@ -3,6 +3,7 @@
 
 include $(top_srcdir)/Makefile.incl
 
+AM_LDFLAGS +=  -L../lib/.libs
 if ENABLE_CLIENT_RELEASE
   AM_LDFLAGS += -static-libtool-libs
 ## for an entirely statically linked library, you may want to try
