Index: boinc/lib/msg_log.cpp
===================================================================
--- boinc.orig/lib/msg_log.cpp	2012-04-01 20:19:44.977682134 +0200
+++ boinc/lib/msg_log.cpp	2012-04-21 11:02:28.079938444 +0200
@@ -19,7 +19,6 @@
 #include "boinc_win.h"
 #else
 #include "config.h"
-#include <cassert>
 #include <cstring>
 #include <string>
 #endif
@@ -70,19 +69,16 @@
     strcpy(spaces+1, "                                                                              ");
 }
 
-void MSG_LOG::enter_level(int diff) {
-    assert (indent_level >= 0);
-
+void MSG_LOG::enter_level(const int diff) {
     if (indent_level <= 0 ) indent_level = 0;
-    if ((indent_level + diff) <= 0) return;
-    if (indent_level >= 39 ) indent_level = 39;
-    if ((indent_level + diff) >= 39) return;
+    if ((indent_level + 2*diff) <= 0) return;
+    if (indent_level >= sizeof(spaces)-1-1 ) indent_level = sizeof(spaces)-1-1;
+    if ((indent_level + 2*diff) >= sizeof(spaces)-1-1) return;
 
     spaces[indent_level] = ' ';
+    spaces[indent_level+1] = 0;
     indent_level += diff*2;
     spaces[indent_level] = 0;
-
-    assert (indent_level >= 0);
 }
 
 void MSG_LOG::vprintf(int kind, const char* format, va_list va) {
Index: boinc/lib/msg_log.h
===================================================================
--- boinc.orig/lib/msg_log.h	2012-04-01 20:19:44.977682134 +0200
+++ boinc/lib/msg_log.h	2012-04-21 11:03:00.187499998 +0200
@@ -50,7 +50,7 @@
     MSG_LOG(FILE* output);
     virtual ~MSG_LOG(){}
 
-    void enter_level(int = 1);
+    void enter_level(const int diff = 1);
     void leave_level() { enter_level(-1); }
     MSG_LOG& operator++() { enter_level(); return *this; }
     MSG_LOG& operator--() { leave_level(); return *this; }
