Index: boinc/lib/coproc.cpp
===================================================================
--- boinc.orig/lib/coproc.cpp	2012-06-30 21:59:10.012848816 +0200
+++ boinc/lib/coproc.cpp	2012-06-30 22:00:39.043650160 +0200
@@ -480,7 +480,7 @@
     display_driver_version = 0;
     strcpy(prop.name, "");
     prop.deviceHandle = 0;
-    prop.totalGlobalMem = 0;
+    prop.totalGlobalMem = 0.0;
     prop.sharedMemPerBlock = 0;
     prop.regsPerBlock = 0;
     prop.warpSize = 0;
@@ -510,7 +510,7 @@
     while (!xp.get_tag()) {
         if (xp.match_tag("/coproc_cuda")) {
             if (!peak_flops) {
-				set_peak_flops();
+                set_peak_flops();
             }
             if (!available_ram) {
                 available_ram = prop.totalGlobalMem;
