--- boinc-7.0.38+dfsg.orig/client/check_security.cpp
+++ boinc-7.0.38+dfsg/client/check_security.cpp
@@ -40,9 +40,9 @@ bool IsUserInGroupBM();
 #endif
 
 static int CheckNestedDirectories(
-    char * basepath, int depth, 
+    const char * const basepath, int depth, 
     int use_sandbox, int isManager, 
-    char * path_to_error
+    const char * const path_to_error
 );
 
 #if (! defined(__WXMAC__) && ! defined(_MAC_INSTALLER))
@@ -501,9 +501,9 @@ saverName[2] = "Progress Thru Processors
 }
 
 
-static int CheckNestedDirectories(char * basepath, int depth,
+static int CheckNestedDirectories(const char * const basepath, int depth,
                                     int use_sandbox, int isManager, 
-                                    char * path_to_error
+                                    const char * const path_to_error
                                 ) {
     int             isDirectory;
     char            full_path[MAXPATHLEN];
@@ -513,6 +513,7 @@ static int CheckNestedDirectories(char *
     dirent          *dp;
     static int      errShown = 0;
 
+    errno=0;
     dirp = opendir(basepath);
     if (dirp == NULL) {
         // Ideally, all project-created subdirectories under project or slot 
@@ -522,6 +523,7 @@ static int CheckNestedDirectories(char *
         // the subdirectory to check them.
         strlcpy(full_path, basepath, sizeof(full_path));
         if ((depth > 1) && (errno == EACCES)) {
+            fprintf(stderr,"Could not open directory %s in client/check_security.cpp: CheckNestedDirectories: %s\n",full_path,strerror(errno));
             return 0;
         } else {
             retval = -1200;
--- boinc-7.0.38+dfsg.orig/client/cs_platforms.cpp
+++ boinc-7.0.38+dfsg/client/cs_platforms.cpp
@@ -45,6 +45,7 @@ LPFN_ISWOW64PROCESS fnIsWow64Process;
 #include <sys/sysctl.h>
 #endif
 
+#include <cerrno>
 #include "error_numbers.h"
 #include "filesys.h"
 #include "str_util.h"
@@ -180,9 +181,17 @@ void CLIENT_STATE::detect_platforms() {
                 int i;
                 for (i=0; i < nlibdirs; i++) {
                     struct dirent *entry;
+                    errno=0;
                     DIR *a = opendir(libdir[i]);
-                    // if dir doesn't exist, do to the next one
-                    if (a == 0) continue;
+                    if (a == 0) {
+                      if (ENOENT==errno) {
+                          // if dir doesn't exist, do to the next one
+                       }
+                      else {
+                         fprintf(stderr,"Unexpected problem with directory '%s': %s\n",libdir[i],strerror(errno));
+                      }
+                      continue;
+                   }
                     // dir exists. read each entry until you find a 32bit lib
                     while ((support32 == 0) && ((entry=readdir(a)) != 0)) {
                         strlcpy(cmdline, file[eno], 256);
