Author: Steffen Moeller <moeller@debian.org>
Description: Improved memory handling.

Index: boinc/lib/parse.cpp
===================================================================
--- boinc.orig/lib/parse.cpp	2012-03-27 16:38:37.000000000 +0200
+++ boinc/lib/parse.cpp	2012-03-27 16:41:59.678459719 +0200
@@ -179,7 +179,12 @@
         int n = (int)strlen(line);
         if (nused + n >= bufsize) {
             bufsize *= 2;
-            buf = (char*)realloc(buf, bufsize);
+            char *b = buf;
+            buf = (char*)realloc(b, bufsize);
+            if (!buf) {
+                free(b); // b was buf and was allocated in this fun
+                return ERR_XML_PARSE;
+            }
         }
         strcpy(buf+nused, line);
         nused += n;
Index: boinc/lib/procinfo_win.cpp
===================================================================
--- boinc.orig/lib/procinfo_win.cpp	2012-03-27 16:38:02.000000000 +0200
+++ boinc/lib/procinfo_win.cpp	2012-03-27 16:41:59.682459664 +0200
@@ -24,13 +24,16 @@
     tNTQSI   pNTQSI = (tNTQSI)GetProcAddress(hNTDllLib, "NtQuerySystemInformation");
     ULONG    cbBuffer = 0;
 
+    if (!ppBuffer) return ERR_NULL;
+    if (!pcbBuffer) return ERR_NULL;
+
     while (1) {
         // Store the buffer size since it appears that somebody is monkeying around
         //   with the return values on some systems.
         cbBuffer = *pcbBuffer;
 
         *ppBuffer = HeapAlloc(hHeap, HEAP_ZERO_MEMORY, *pcbBuffer);
-        if (ppBuffer == NULL) {
+        if (*ppBuffer == NULL) {
             return ERR_MALLOC;
         }
 
@@ -54,9 +57,9 @@
         } else if (!NT_SUCCESS(Status)) {
             HeapFree(hHeap, NULL, *ppBuffer);
             return ERR_GETRUSAGE;
-		} else {
-			return 0;
-		}
+        } else {
+            return 0;
+        }
     }
     return 0;	// never reached
 }
