Index: boinc/tools/appmgr
===================================================================
--- boinc.orig/tools/appmgr	2011-06-20 21:38:42.000000000 +0200
+++ boinc/tools/appmgr	2011-07-24 22:46:49.000000000 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python
 #
 # Written by Gabor Gombas <gombasg@sztaki.hu>
 # Licensed under the same terms as the rest of BOINC.
Index: boinc/tools/dbcheck_files_exist
===================================================================
--- boinc.orig/tools/dbcheck_files_exist	2011-06-20 21:38:42.000000000 +0200
+++ boinc/tools/dbcheck_files_exist	2011-07-24 22:46:49.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 import os, re, boinc_path_config
 from Boinc import database, db_mid, boinc_db, configxml
Index: boinc/tools/make_project
===================================================================
--- boinc.orig/tools/make_project	2011-07-24 19:39:10.000000000 +0200
+++ boinc/tools/make_project	2011-07-24 22:46:49.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # $Id: make_project 23509 2011-05-06 12:33:12Z davea $
 # Creates a new BOINC project.
Index: boinc/tools/parse_config
===================================================================
--- boinc.orig/tools/parse_config	2011-06-20 21:38:42.000000000 +0200
+++ boinc/tools/parse_config	2011-07-24 22:46:49.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # -*- mode: python; python-indent: 4; -*-
 
Index: boinc/tools/upgrade
===================================================================
--- boinc.orig/tools/upgrade	2011-07-24 19:39:10.000000000 +0200
+++ boinc/tools/upgrade	2011-07-24 22:46:49.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # $Id: upgrade 23358 2011-04-09 03:10:29Z davea $
 
Index: boinc/tools/xadd
===================================================================
--- boinc.orig/tools/xadd	2011-06-20 21:38:42.000000000 +0200
+++ boinc/tools/xadd	2011-07-24 22:46:49.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # $Id: xadd 18374 2009-06-11 15:38:48Z davea $
 
