Index: boinc/sched/sched_driver.cpp
===================================================================
--- boinc.orig/sched/sched_driver.cpp	2012-03-27 14:29:39.610419635 +0200
+++ boinc/sched/sched_driver.cpp	2012-03-27 14:48:59.970588289 +0200
@@ -45,13 +45,14 @@
 #include "util.h"
 #include "svn_version.h"
 #include "str_util.h"     // safe_strcpy
+#include "str_replace.h"     // strlcpy
 
 using std::vector;
 
 struct HOST_DESC{
-    char os_name[128];
-    char p_vendor[128];
-    char p_model[128];
+    char os_name[512];
+    char p_vendor[512];
+    char p_model[512];
 };
 
 vector<HOST_DESC> host_descs;
@@ -59,7 +60,7 @@
 double max_time = 1;
 
 void read_hosts() {
-    char buf[256], buf2[256];
+    char buf[1024], buf2[1024];
     host_descs.clear();
     FILE* f = fopen("host_descs.txt", "r");
     if (!f) {
