Index: boinc/stripchart/stripchart.cnf
===================================================================
--- boinc.orig/stripchart/stripchart.cnf	2011-07-24 20:08:48.000000000 +0200
+++ boinc/stripchart/stripchart.cnf	2011-07-24 20:11:18.000000000 +0200
@@ -8,7 +8,7 @@
 # Directory where gnuplot is
 # $gnuplot = "/usr/local/gnuplot-3.7";
 #$gnuplot = "/disks/asimov/a/users/hiramc/local/src/gnuplot-3.7";
-$gnuplot = "/usr/local/bin"
+$gnuplot = "/usr/bin"
  
 # Temporary files
 $suffix = rand(10000);
@@ -54,9 +54,9 @@
 # Note: there is a sample copy in the samples directory: 
 $datafilelist = "datafiles";
  
-# Where is the list of user-definied cgi queries?
+# Where is the list of user-defined cgi queries?
 # Note: this file gets created by the cgi - must put it somewhere that the cgi user can write to 
-$queryfilelist = "querylist";
+$queryfilelist = "/tmp/querylist";
 
 # What time is it right now?
 $rightnow = time;
