Index: boinc/api/Makefile.am
===================================================================
--- boinc.orig/api/Makefile.am	2011-07-24 23:28:34.000000000 +0200
+++ boinc/api/Makefile.am	2011-07-24 23:30:06.000000000 +0200
@@ -16,11 +16,11 @@
     gutil.cpp \
     gutil_text.cpp \
     reduce_lib.cpp \
-    texfont.cpp \
     texture.cpp \
-    txf_util.cpp \
     graphics2.cpp \
     graphics2_unix.cpp
+    #texfont.cpp \
+    #txf_util.cpp \
 
 if OS_DARWIN
     graphics2_files += mac_icon.cpp
Index: boinc/api/txf_util.cpp
===================================================================
--- boinc.orig/api/txf_util.cpp	2011-07-24 23:28:53.000000000 +0200
+++ boinc/api/txf_util.cpp	2011-07-24 23:29:47.000000000 +0200
@@ -26,14 +26,15 @@
 #include "filesys.h"
 #include "boinc_api.h"
 
-#include "texfont.h"
+//#include "texfont.h"
 #include "txf_util.h"
 
 #ifndef _MAX_PATH
 #define _MAX_PATH 256
 #endif 
 
-static TexFont* txf[TXF_NUM_FONT];
+//static TexFont* txf[TXF_NUM_FONT];
+static void* txf[TXF_NUM_FONT];
 
 // load fonts. call once.
 //
Index: boinc/api/txf_util.h
===================================================================
--- boinc.orig/api/txf_util.h	2011-07-24 23:28:53.000000000 +0200
+++ boinc/api/txf_util.h	2011-07-24 23:29:47.000000000 +0200
@@ -21,6 +21,8 @@
 #ifndef _TXF_UTIL_
 #define _TXF_UTIL_
 
+#include <GL/gl.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
