Author: Steffen Moeller <moeller@debian.org>
Description: Addressing Valgrind reporting lacking initialisations of BOINC.

Index: boinc/client/client_types.h
===================================================================
--- boinc.orig/client/client_types.h	2012-03-31 02:15:34.846451729 +0200
+++ boinc/client/client_types.h	2012-03-31 02:15:35.074448621 +0200
@@ -727,7 +727,17 @@
     WORKUNIT* wup;
     PROJECT* project;
 
-    RESULT(){}
+    RESULT(){
+       edf_scheduled=0;
+       coproc_missing=0;
+       suspended_via_gui=0;
+       report_immediately=0;
+       not_started=0;
+       project=NULL;
+       wup=NULL;
+       app=NULL;
+       clear();
+    }
     ~RESULT(){}
     void clear();
     int parse_server(XML_PARSER&);
