Index: boinc/tools/process_input_template.cpp
===================================================================
--- boinc.orig/tools/process_input_template.cpp	2012-05-10 12:51:41.351949895 +0200
+++ boinc/tools/process_input_template.cpp	2012-05-10 12:52:24.000000000 +0200
@@ -17,6 +17,7 @@
 
 #include <stdio.h>
 #include <string>
+#include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
Index: boinc/sched/single_job_assimilator.cpp
===================================================================
--- boinc.orig/sched/single_job_assimilator.cpp	2012-03-31 13:44:26.867113620 +0200
+++ boinc/sched/single_job_assimilator.cpp	2012-05-10 13:05:44.388571200 +0200
@@ -24,6 +24,7 @@
 #include <string>
 #include <cstdlib>
 #include <cstring>
+#include <unistd.h>
 
 #include "boinc_db.h"
 #include "error_numbers.h"
Index: boinc/tools/create_work.cpp
===================================================================
--- boinc.orig/tools/create_work.cpp	2012-03-31 13:44:27.107110342 +0200
+++ boinc/tools/create_work.cpp	2012-05-10 13:06:35.687877523 +0200
@@ -26,6 +26,7 @@
 #include <cstring>
 #include <ctime>
 #include <string>
+#include <unistd.h>
 
 #include "backend_lib.h"
 #include "boinc_db.h"
Index: boinc/vda/vda.cpp
===================================================================
--- boinc.orig/vda/vda.cpp	2012-03-31 13:47:01.892995979 +0200
+++ boinc/vda/vda.cpp	2012-05-10 13:07:12.487379494 +0200
@@ -21,6 +21,7 @@
 // vda add path     add a file
 
 #include <stdio.h>
+#include <unistd.h>
 
 #include "boinc_db.h"
 #include "filesys.h"
Index: boinc/vda/vdad.cpp
===================================================================
--- boinc.orig/vda/vdad.cpp	2012-04-19 18:05:13.228502956 +0200
+++ boinc/vda/vdad.cpp	2012-05-10 13:09:12.785751379 +0200
@@ -27,6 +27,7 @@
 #include <sys/types.h>
 #include <vector>
 #include <set>
+#include <unistd.h>
 
 using std::vector;
 using std::set;
