--- a/tools/process_input_template.cpp
+++ b/tools/process_input_template.cpp
@@ -17,6 +17,7 @@
 
 #include <stdio.h>
 #include <string>
+#include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
--- a/sched/single_job_assimilator.cpp
+++ b/sched/single_job_assimilator.cpp
@@ -24,6 +24,7 @@
 #include <string>
 #include <cstdlib>
 #include <cstring>
+#include <unistd.h>
 
 #include "boinc_db.h"
 #include "error_numbers.h"
--- a/tools/create_work.cpp
+++ b/tools/create_work.cpp
@@ -27,6 +27,7 @@
 #include <cstring>
 #include <ctime>
 #include <string>
+#include <unistd.h>
 
 #include "backend_lib.h"
 #include "boinc_db.h"
--- a/vda/vda.cpp
+++ b/vda/vda.cpp
@@ -21,6 +21,7 @@
 // vda add path     add a file
 
 #include <stdio.h>
+#include <unistd.h>
 
 #include "boinc_db.h"
 #include "filesys.h"
--- a/vda/vdad.cpp
+++ b/vda/vdad.cpp
@@ -27,6 +27,7 @@
 #include <sys/types.h>
 #include <vector>
 #include <set>
+#include <unistd.h>
 
 using std::vector;
 using std::set;
