Index: boinc/tools/update_versions
===================================================================
--- boinc.orig/tools/update_versions	2011-06-12 20:36:24.000000000 +0200
+++ boinc/tools/update_versions	2011-06-12 20:36:37.000000000 +0200
@@ -16,7 +16,7 @@
 
 import boinc_path_config
 from Boinc import database, db_mid, configxml, tools, boinc_project_path
-import sys, os, re, time, string
+import sys, os, re, time, string, tempfile
 from optparse import OptionParser
 
 parser = OptionParser(usage="usage: %prog [options]")
@@ -54,19 +54,16 @@
     return map(lambda file: os.path.join(dir, file), os.listdir(dir))
 
 def get_api_version(exec_file):
-    tmpfile = '.uvtemp'
-    cmd = "strings %s | grep API_VERSION > %s"%(exec_file, tmpfile)
+    tmpfile = tempfile.NamedTemporaryFile()
+    cmd = "strings %s | grep API_VERSION > %s"%(exec_file, tmpfile.name)
     os.system(cmd)
-    f = open(tmpfile, 'r')
-    if (f):
-        s = string.strip(f.read())
-        f.close()
-        os.unlink(tmpfile)
-        prefix = 'API_VERSION_'
-        n = string.find(s, prefix)
-        if (n == 0):
-            k = len(prefix)
-            return s[k:]
+    s = string.strip(tmpfile.read())
+    tmpfile.close()
+    prefix = 'API_VERSION_'
+    n = string.find(s, prefix)
+    if (n == 0):
+        k = len(prefix)
+        return s[k:]
     return ''
 
 # we've scanned an app version directory or file.
