Author: Steffen Moeller <moeller@debian.org>
Description: About nothing should be world writable.
Index: boinc/test/test.inc
===================================================================
--- boinc.orig/test/test.inc	2011-06-12 20:42:41.000000000 +0200
+++ boinc/test/test.inc	2011-06-12 20:43:04.000000000 +0200
@@ -348,9 +348,9 @@
     function mkdir($dir, $chmod=0)
     {
         $d = "$this->project_dir/$dir";
-        mkdir($d,0777);
+        mkdir($d,0775);
         if ($chmod) {
-            chmod($d, 0777);
+            chmod($d, 0775);
         }
     }
 
