Index: boinc/clientgui/BOINCGUIApp.h
===================================================================
--- boinc.orig/clientgui/BOINCGUIApp.h	2012-05-07 00:45:29.061521189 +0200
+++ boinc/clientgui/BOINCGUIApp.h	2012-05-07 00:46:34.000000000 +0200
@@ -117,7 +117,7 @@
 
 public:
 
-    bool                OnInit();
+    virtual bool        OnInit();
 
     wxLocale*           GetLocale()                 { return m_pLocale; }
     CSkinManager*       GetSkinManager()            { return m_pSkinManager; }
